### PR TITLE
Fixed includeTasks in get workflows by correlationId. Earlier, Tasks …

### PR DESCRIPTION
**Pull request by kishorebanala**
…are included by default. Now, includeTasks should direct tasks presence in Workflow results.

### CODE REVIEW

File: core/src/main/java/com/netflix/conductor/dao/ExecutionDAO.java [hunk 1]
```diff
@@ -218,15 +219,27 @@
...
+
+
+	/**
+	 *
+	 * @param workflowId
+	 * @return List of Tasks for given Workflow, in sorted order.
+	 */
+	default List<Task> getWorkflowTasksSorted(String workflowId) {
+		List<Task> tasks = getTasksForWorkflow(workflowId);
+		tasks.sort(Comparator.comparingLong(Task::getScheduledTime).thenComparingInt(Task::getSeq));
```
Comment (cyzhao) [inline core/src/main/java/com/netflix/conductor/dao/ExecutionDAO.java]: Is there a need to sort it?

File: core/src/main/java/com/netflix/conductor/service/ExecutionService.java [hunk 2]
```diff
@@ -280,33 +280,29 @@ private boolean requeue(Task pending) throws Exception {
...
-				try {
-					tasks = edao.getTasksForWorkflow(wf.getWorkflowId());
-				} catch (Exception e) {
-					throw new RuntimeException(e);
-				}
-				wf.setTasks(tasks);
-			});
+	public List<Workflow> getWorkflowInstances(String workflowName, String correlationId, boolean includeClosed, boolean includeTasks) {
+
+		List<Workflow> workflows = executionDAO.getWorkflowsByCorrelationId(correlationId, includeTasks);
```
Comment (cyzhao) [inline core/src/main/java/com/netflix/conductor/service/ExecutionService.java]: This is duplicated with WorkflowExecutor.getStatusByCorrelationId after another change above, why not just call it here?

File: core/src/main/java/com/netflix/conductor/dao/ExecutionDAO.java (hunk 1 above)
Comment (cyzhao) [reply core/src/main/java/com/netflix/conductor/dao/ExecutionDAO.java]: never mind, this is existing code, just moved here.

File: mysql-persistence/src/main/java/com/netflix/conductor/dao/mysql/MySQLExecutionDAO.java [hunk 3]
```diff
@@ -367,13 +365,13 @@ public long getInProgressTaskCount(String taskDefName) {
...
 	@Override
-	public List<Workflow> getWorkflowsByCorrelationId(String correlationId) {
+	public List<Workflow> getWorkflowsByCorrelationId(String correlationId, boolean includeTasks) {
 		Preconditions.checkNotNull(correlationId, "correlationId cannot be null");
 		String GET_WORKFLOWS_BY_CORRELATION_ID = "SELECT workflow_id FROM workflow WHERE correlation_id = :correlationId";
 		return getWithTransaction(tx -> tx.createQuery(GET_WORKFLOWS_BY_CORRELATION_ID)
 				.addParameter("correlationId", correlationId)
 				.executeScalarList(String.class)).stream()
-				.map(this::getWorkflow)
+				.map(wfid -> getWorkflow(wfid, includeTasks))
```
Comment (cyzhao) [inline mysql-persistence/src/main/java/com/netflix/conductor/dao/mysql/MySQLExecutionDAO.java]: wfid -> workflowId

File: redis-persistence/src/main/java/com/netflix/conductor/dao/dynomite/RedisExecutionDAO.java [hunk 4]
```diff
@@ -362,24 +362,19 @@ public Workflow getWorkflow(String workflowId) {
 	@Override
 	public Workflow getWorkflow(String workflowId, boolean includeTasks) {
```
Comment (cyzhao) [inline redis-persistence/src/main/java/com/netflix/conductor/dao/dynomite/RedisExecutionDAO.java]: Same issue with MySQLExecutionDAO, please revert this changes.

File: core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java [hunk 5]
```diff
@@ -297,20 +297,6 @@ public void retry(String workflowId) throws Exception {
 
     }
 
-    public List<Workflow> getStatusByCorrelationId(String workflowName, String correlationId, boolean includeClosed) throws Exception {
```
Comment (cyzhao) [inline core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java]: Did you intend to remove this method?

Comment (kishorebanala) [reply core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java]: Yes, it is not used anywhere else, and this functionality is replicated in ExecutionService.
